app/workers/articles/update_organic_page_views_worker.rb

```diff
@@ -0,0 +1,37 @@
+module Articles
+  class UpdateOrganicPageViewsWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :medium_priority,
+                    lock: :until_executing,
+                    on_conflict: :replace,
+                    retry: 10
+
+    GOOGLE_REFERRER = "https://www.google.com/".freeze
+
+    def perform(article_id)
+      article = Article.find(article_id)
```

Comment:
This will fail if the article ID is not present, AFAIK we tend to `return unless article` to avoid having retrying jobs on resources that do not exist anymore

Reply:
Thanks for the reminder @rhymes, the silent swallowing of errors in jobs is something that I'll never quite get used to it seems.

Reply:
Yes, this is true. As you know, I was never a big fan of us silently swallowing errors in jobs in this way. IMHO the best approach is to let them raise, catch them, then manually report them to some error tracking service and having the job successfully finish without retries. 

```ruby
def perform(some_id)
  some_model = SomeModel.find(some_id)
  # code working on some_model
rescue ActiveRecord::RecortNotFound => e
  # Notify Honeybadger or other service
end
```

This way we're made aware that there was a problem while also saving on futile retries.

It's also by far the most consistent approach: elsewhere in this PR, you suggested using `create!` over `create`, but wouldn't this also result in a futile retry? If the params weren't valid the first time around, they still won't be valid on retry. Now of course there could have been a temporary DB issue, but that seems a bit unlikely and would be remedied by the next run of this job anyway.

With all that said, I decided that the RightThing™ for these two specific jobs is changing the jobs from `retry: 10` to `retry: false`  for the following reasons:

1. We may want to be notified of errors here.
2. I don't see how this would fail in a way where a retry makes sense.

What do you think?

Reply:
> Yes, this is true. As you know, I was never a big fan of us silently swallowing errors in jobs in this way. IMHO the best approach is to let them raise, catch them, then manually report them to some error tracking service and having the job successfully finish without retries.

I'd see this as an argument for a refactoring, in the meantime I'd stick with the current approach

> It's also by far the most consistent approach: elsewhere in this PR, you suggested using create! over create, but wouldn't this also result in a futile retry? If the params weren't valid the first time around, they still won't be valid on retry. Now of course there could have been a temporary DB issue, but that seems a bit unlikely.

Yeah, I was wrong, it's better to test if the object was actually created then :D

> I don't see how this would fail in a way where a retry makes sense.

Makes sense. As we know it'll eventually be updated the moment you open the article and we don't want hundreds of failed jobs trying to update their page views. It's better for it to fail and us figuring out why they keep failing if they do

Reply:
> I'd see this as an argument for a refactoring, in the meantime I'd stick with the current approach

This is a sentiment I don't understand. Consistency is good, but IMHO shouldn't be a reason for continuing a pattern that doesn't seem particularly useful* as that just leaves more to be refactored later. 

If none of our migrations had ever added an index, we wouldn't continue doing that but instead start adding them in new ones and then slowly add the missing ones, wouldn't we?

* I actually consider it actively harmful as it sets a bad precedent for silently ignoring errors.

> Makes sense.

Just to make 100% sure, does this mean your ok with the approach taken in these two workers?

Reply:
I'm okay :-) For all the reasons you just said. If something doesn't work and you argumented eloquently, let's change it. I still think we might need a RFC to change the approach for existing jobs, at least to have a "piece of paper" that says: this is what we're doing wrong, this is what we're going to do to change it.

Reply:
Mostly because changing how we process jobs and what we consider a failure might have an impact on SRE and on call procedures

Reply:
Yes, that's a good point and I will put this on my RFC to do list.

Reply:
Keep in mind we have a new Principal SRE starting shortly. @citizen428 if you don't have the bandwidth to write an RFC to overhaul our worker patterns feel free to drop a note about it [on the SRE board](https://github.com/orgs/forem/projects/6) and let Mac know. Might be a good little getting started task for the new SRE.

Reply:
@mstruve 👌 great idea.